valmlord/gulpify
# feat: add favicon generation  task

## Description
This pull request enhances the Gulp boilerplate workflow by adding a task specifically designed for generating favicons from a single source image. It includes the integration of `gulp-favicons` and the configuration of paths for source and output directories.

## Changes
- Added `gulp-favicons` package to `package.json`.
- Created a new Gulp task to generate favicons using `gulp-favicons`.
- Configured the favicon generation process with appropriate settings for app name, description, developer information, and display options.
- Updated `paths.js` to include paths for the source favicon image and the output directory.
- Integrated the new favicon task into the main Gulp workflow.

## How to Test
1. Place the source favicon image in the appropriate directory `src/assets/icons/favicon.png`.
2. Execute the relevant Gulp task `gulp faviconsTask` to generate the favicons.
3. Verify that the generated favicons are correctly created in the `dist/assets/favicons` directory.
4. Include the generated favicons in your HTML or Pug templates by referencing the generated HTML file or directly linking to the favicons.

## Additional Notes
- This update aims to improve the efficiency of handling favicons by generating all necessary icons for various devices and browsers from a single source image.
- The configuration ensures that the favicons are correctly displayed across different platforms and devices, providing a consistent user experience.
